File: athrill-account-client/projects/at-models/src/lib/basics/teams/team-category.model.ts
import { BaseModel,BaseAddModel,BaseUpdateModel } from '../../at-base.model';
import { TeamCategoryMemberLinkModel } from './team-category-member-link.model';

export class TeamCategoryModel extends BaseModel{
    public TeamCategoryId: number;
    public Orientation: number;
    public Name: string;;
    public ShortDescription: string;
    public Description: string;
    public IsPublished: boolean;
    public Id: number;

    public TeamCategoryMemberLinks : TeamCategoryMemberLinkModel[];
}

export class TeamCategoryAddModel extends BaseAddModel{
    public Name: string;;
    public ShortDescription: string;
    public Description: string;
    public IsPublished: boolean;

    public TeamCategoryMemberLinks : TeamCategoryMemberLinkModel[];
}

export class TeamCategoryUpdateModel extends BaseUpdateModel{
    public TeamCategoryId: number;
    public Name: string;;
    public ShortDescription: string;
    public Description: string;
    public IsPublished: boolean;
}

File: athrill-account-server/Libraries/Model/Basics/Teams/TeamCategoryModel.cs
using FluentValidation;
using System;
using System.Collections.Generic;
using System.Text;

namespace AT.Model.Basics.Teams
{
    public class TeamCategoryModel : BaseModel
    {
        public int TeamCategoryId { get; set; }
        public int Orientation { get; set; }
        public string Name { get; set; }
        public string ShortDescription { get; set; }
        public string Description { get; set; }
        public bool IsPublished { get; set; }

        #region Override

        public override int Id => TeamCategoryId;
        public override string ToString()
        {
            return Name;
        }

        #endregion

        #region Links

        public ICollection<TeamCategoryMemberLinkModel> TeamCategoryMemberLinks { get; set; }

        #endregion
    }

    public class TeamCategoryAddModel : BaseAddModel
    {
        public string Name { get; set; }
        public string ShortDescription { get; set; }
        public string Description { get; set; }
        public bool IsPublished { get; set; }
    }

    public class TeamCategoryUpdateModel : BaseUpdateModel
    {
        public int TeamCategoryId { get; set; }
        public string Name { get; set; }
        public string ShortDescription { get; set; }
        public string Description { get; set; }
        public bool IsPublished { get; set; }
    }

    public class TeamCategoryAddModelValidator : AbstractValidator<TeamCategoryAddModel>
    {
        public TeamCategoryAddModelValidator()
        {
            RuleFor(m => m.Name).NotEmpty().NotNull().Length(1, 200);
            RuleFor(m => m.ShortDescription).Length(1, 600);
            RuleFor(m => m.IsPublished).NotNull();
        }
    }

    public class TeamCategoryUpdateModelValidator : AbstractValidator<TeamCategoryUpdateModel>
    {
        public TeamCategoryUpdateModelValidator()
        {
            RuleFor(m => m.Name).NotEmpty().NotNull().Length(1, 200);
            RuleFor(m => m.ShortDescription).Length(1, 600);
            RuleFor(m => m.IsPublished).NotNull();
        }
    }
}
